examples/pwelch.cu

```diff
@@ -0,0 +1,101 @@
+////////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (c) 2023, NVIDIA Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#include "matx.h"
+#include <cassert>
+#include <cstdio>
+#include <cuda/std/ccomplex>
+
+using namespace matx;
+
+/**
+ * PWelch Example
+ *
+ * This example shows how to estimate the power spectral density of a signal using the pwelch() operator
+ * using Welch's method.
+ *
+ */
+
+int main([[maybe_unused]] int argc, [[maybe_unused]] char **argv)
+{
+  MATX_ENTER_HANDLER();
+  using complex = cuda::std::complex<float>;
+
+  float exec_time_ms;
+  const int num_iterations = 100;
+  index_t signal_size = 256;
+  index_t nperseg = 32;
+  index_t nfft = nperseg;
+  index_t noverlap = 8;
+  float ftone = 3.0;
+  cudaStream_t stream;
+  cudaStreamCreate(&stream);
+  cudaEvent_t start, stop;
+  cudaEventCreate(&start);
+  cudaEventCreate(&stop);
+
+  // Create buffers
+  auto x  = make_tensor<complex>({signal_size});
+  auto Pxx  = make_tensor<typename complex::value_type>({nfft});
+
+  for (index_t k = 0; k < signal_size; k++) {
```

Comment:
your 2.0 should be 2.0f.
 
 we should also consider an operator implementation.  I don't think we quite have what we need here though.  ideally we would have a sincos operator which takes a real and outputs complex.   @cliffburdick for thoughts.

Reply:
If you use 2.0f you should also cast M_PI to cast to a float or it's going to do double math anyways. This is just the example setup, so I'm not sure it matters.

Since this is the initialization code, I also don't have a strong opinion on changing this to an operator. If you wanted to, you could make a `sincosf` and `sincospif`  and pass in a `phase` vector to compute it.

Reply:
I can make the 2.0f and M_PI casting change now.  Do you want to include a complex exponential generator operator in this PR, or leave that for a separate PR?

Reply:
Pushed a new commit using the existing expj() operator...